Adding page for istio image signing and validation (#10775)
istio/istio.io: content/en/docs/ops/best-practices/image-signing-validation/index.md

---
title: Image Signing And Validation
description: Describes how to use image signatures to verify the provenance of Istio images.
weight: 35
aliases:

>     - /docs/ops/best-practices/image-signing-validation/

Reviewer: Not sure we need an alias for a new page.

Author: Removed

> keywords: [install]

Reviewer: Do we want some more keywords, like `signing`? Not sure what would make sense.

Author: Added one, but also not sure what else makes sense.

> owner: istio/wg-environments-maintainers

Reviewer: Should environments own this page?

Author: I wasn't sure, here. AFAIK no existing pages are owned by test and release, and it wasn't clear if this was a group that was internal to the istio.io repo or if it was referencing github org groups.

Reviewer: It would refer to the WG that would approve changes to this page (or be responsible if a test suddenly started failing (noting no test on this page so can ignore that)). A doc reviewer, may not understand the technicalities of the page so would expect that someone from the specified WG would approve the PR before it would be approved by the doc reviewer and merge.

Reviewer 2: The environments group will take ownership of this page. That's fine with me.

Author: Leaving as Environments, then.

> test: n/a

Reviewer: Might be interesting to actually add a test for this page if we don't actively check that the commands work elsewhere.

---

This page describes how to use [Cosign](https://github.com/sigstore/cosign) to
validate the provenance of Istio image artifacts.

Cosign is a tool developed as part of the
[sigstore](https://www.sigstore.dev/https://www.sigstore.dev/) project which

> simplifies signing and validation of signed Open Container Initiative artifacts,

Reviewer 2: ```suggestion
simplifies signing and validation of signed Open Container Initiative (OCI) artifacts,
```

such as docker images.

In Istio, we sign all officially published docker images as part of our release

> process using our own private key. End users can then verify these images using

Reviewer 2: ```suggestion
process using our own private key starting with Istio 1.12. End users can then verify these images using
```

Reviewer: We may actually be able to say 1.11.x (not sure what x is).

the process described below.

This process is suitable for either manual execution or integration with build
or deployment pipelines for automated verification of artifacts.

## Prerequisites

Before you begin, please do the following:

1. Download the latest
   [Cosign](https://github.com/sigstore/cosign/releases/latest) build for your
   architecture, as well as its signature.
1. Validate the cosign binary signature: 
   {{< text bash >}}
$ openssl dgst -sha256 \
	-verify <(curl -ssL https://raw.githubusercontent.com/sigstore/cosign/main/release/release-cosign.pup) \
	-signature /path/to/cosign.sig \
	/path/to/cosign-binary
    {{< /text >}}
1. Run `chmod +x` to make the Cosign binary executable

## Validating Image

To validate a docker image, do the following:

{{< text bash >}}

> $ ./cosign-binary verify --key {{< static "/misc/istio-key.pub" >}} {{< istio_docker_image "pilot" >}}

Reviewer: The pub key isn't working correctly, at least if I look at the deploy/netlify test details link, https://deploy-preview-10775--preliminary-istio.netlify.app/latest/docs/ops/best-practices/image-signing-validation/

Author: It looks like it is setting `$.Site.Page.BaseURL` to `/latest/` rather than the full URL.

{{< /text >}}

> This process will work for any released image or release candidate built with Istio build infrastructure.

Reviewer: with the

Author: Done.

---

> title: Image Signing And Validation

Reviewer 3: ```suggestion
title: Image Signing and Validation
```

description: Describes how to use image signatures to verify the provenance of Istio images.
weight: 35
aliases: []
keywords: [install,signing]
owner: istio/wg-environments-maintainers
test: n/a
---

> This page describes how to use [Cosign](https://github.com/sigstore/cosign) to

Reviewer 4: ```suggestion
This page describes how to use [cosign](https://github.com/sigstore/cosign) to
```

> This page describes how to use [cosign](https://github.com/sigstore/cosign) to

Reviewer 3: ```suggestion
This page describes how to use [Cosign](https://github.com/sigstore/cosign) to
```

validate the provenance of Istio image artifacts.

> Cosign is a tool developed as part of the

Reviewer 4: ```suggestion
cosign is a tool developed as part of the
```

> cosign is a tool developed as part of the

Reviewer 3: ```suggestion
Cosign is a tool developed as part of the
```

> [sigstore](https://www.sigstore.dev) project which
> simplifies signing and validation of signed Open Container Initiative (OCI) artifacts,
> such as docker images.

Reviewer 3: ```suggestion
[sigstore](https://www.sigstore.dev) project, which
simplifies signing and validation of signed Open Container Initiative (OCI) artifacts,
such as container images.
```

> In Istio, we sign all officially published docker images as part of our release
> process using our own private key starting with Istio 1.12. End users can then verify these images using

Reviewer 3: ```suggestion
Starting with Istio 1.12, we sign all officially published container images as part of our release
process. End users can then verify these images using
```

the process described below.

This process is suitable for either manual execution or integration with build
or deployment pipelines for automated verification of artifacts.

## Prerequisites

Before you begin, please do the following:

1. Download the latest
   [Cosign](https://github.com/sigstore/cosign/releases/latest) build for your
   architecture, as well as its signature.

> 1. Validate the cosign binary signature:

Reviewer 3: ```suggestion
1. Validate the `cosign` binary signature:
```

   {{< text bash >}}
$ openssl dgst -sha256 \

>     -verify <(curl -ssL https://raw.githubusercontent.com/sigstore/cosign/main/release/release-cosign.pup) \

Reviewer 4: ```suggestion
    -verify <(curl -ssL https://raw.githubusercontent.com/sigstore/cosign/main/release/release-cosign.pub) \
```

>     -signature /path/to/cosign.sig \

Reviewer 4: ```suggestion
    -signature <(cat /path/to/cosign.sig | base64 -d) \
```

    /path/to/cosign-binary
    {{< /text >}}

> 1. Run `chmod +x` to make the Cosign binary executable

Reviewer 4: ```suggestion
1. Make the binary executable (`chmod +x`) and move to a location on the `PATH`
```

## Validating Image

> To validate a docker image, do the following:

Reviewer 4: ```suggestion
To validate a container image, do the following:
```
Suggest using "container image" throughout document
